src/main/java/org/openrewrite/java/testing/junit5/MigrateFromHamcrest.java

```diff
@@ -0,0 +1,119 @@
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            J.MethodInvocation mi = super.visitMethodInvocation(method, executionContext);
+            MethodMatcher matcherAssertTrue = new MethodMatcher("org.hamcrest.MatchAssert assertThat(String, boolean)");
+            MethodMatcher matcherAssertMatcher = new MethodMatcher("org.hamcrest.MatcherAssert assertThat(*, org.hamcrest.Matcher)");
+            MethodMatcher matcherAssertMatcherWithReason = new MethodMatcher("org.hamcrest.MatcherAssert assertThat(String,*,org.hamcrest.Matcher)");
+
+            if (matcherAssertTrue.matches(mi)) {
+                //TODO simple
+            } else if (matcherAssertMatcher.matches(mi)) {
+                Expression hamcrestMatcher = mi.getArguments().get(1);
+                if (hamcrestMatcher instanceof J.MethodInvocation) {
+                    J.MethodInvocation matcherInvocation = (J.MethodInvocation)hamcrestMatcher;
+                    maybeRemoveImport("org.hamcrest.Matchers." + matcherInvocation.getSimpleName());
+                    maybeRemoveImport("org.hamcrest.MatcherAssert.assertThat");
+                    String targetAssertion = getTranslatedAssert(matcherInvocation);
+                    JavaTemplate template = JavaTemplate.builder(this::getCursor, getTemplateForTranslatedAssertion(targetAssertion))
+                      .javaParser(JavaParser.fromJavaVersion().classpathFromResources(executionContext, "junit-jupiter-api-5.9"))
+                      .staticImports("org.junit.jupiter.api.Assertions." + targetAssertion)
+                      .build();
+                    mi = withTemplate(mi, template, mi.getArguments().get(0), stripMatcherInvocation(mi.getArguments().get(1)));
+                    maybeAddImport("org.junit.jupiter.api.Assertions", targetAssertion);
+                }
+                else throw new IllegalArgumentException("Parameter mismatch for " + mi + ".");
+            }
+            return mi;
+        }
+
+        private J.MethodInvocation withTemplate(J.MethodInvocation method, JavaTemplate template, Expression firstArg, List<Expression> matcherArgs) {
+            switch (matcherArgs.size()) {
+                case 0:
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg);
+                case 1:
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0));
+                case 2 :
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0), matcherArgs.get(1));
+                case 3 :
+                    return method.withTemplate(template, method.getCoordinates().replace(), firstArg, matcherArgs.get(0), matcherArgs.get(1), matcherArgs.get(2));
```

Comment:
The last argument in withTemplate is actually a varargs. So you could reduce this when opting for a `Object[] args` as seen here. https://github.com/openrewrite/rewrite-testing-frameworks/blob/f25ff0b0ed2b9e227b07972986b35f7a1cc958af/src/main/java/org/openrewrite/java/testing/cleanup/AssertTrueEqualsToAssertEquals.java#L75-L80